model/database.php

```diff
@@ -95,6 +94,31 @@ function getTutor($year, $userID)
 
     }
 
+    /**
+     * Updates data is the Year table given a column, value and yearId
+     *
+     * @param string $column The name of the column in the database being updated
+     * @param mixed $value The value to set the column to
+     * @param int $yearId The year_id for the year data being updated
+     * @author Keller Flint
+     */
+    function updateYearData($column, $value, $yearId)
+    {
+
+        // TODO validate $column against php array of column names used in db to prevent sql injection
+
+        if ($value == '0' || $value == '1') {
+            $sql = "UPDATE Year SET $column = b? WHERE year_id = ?";
+        } else {
```

Comment:
whats the point of "b" ?

Reply:
You have to use b when adding a binary value cause sql is an idiot and cant interpret '0' or '1' as b'0' and b'1' without explicitly writing it in the statement. So any time I detect a value that is supposed to be binary, it has to have the b prepended to the front of the '0' or '1'.